osu.Game/Screens/SelectV2/BeatmapPanel.cs

```diff
@@ -79,9 +271,127 @@ protected override void PrepareForUse()
             Debug.Assert(Item != null);
             var beatmap = (BeatmapInfo)Item.Model;
 
-            text.Text = $"Difficulty: {beatmap.DifficultyName} ({beatmap.StarRating:N1}*)";
+            iconContainer.Icon = beatmap.Ruleset.CreateInstance().CreateIcon();
+
+            difficultyRank.Beatmap = beatmap;
+            difficultyText.Text = beatmap.DifficultyName;
+            authorText.Text = BeatmapsetsStrings.ShowDetailsMappedBy(beatmap.Metadata.Author.Username);
 
-            this.FadeInFromZero(500, Easing.OutQuint);
+            starDifficultyBindable = null;
+            keyCount.Value = null;
+
+            updateVisibilityState();
+            isCurrentItemVisible.TriggerChange();
```

Comment:
naaaaaaaaaaaaaaaaaaaaaaaaaaaah

Just do the SR calculation immediately, this is all jank as hell (no bindable, no `TriggerChange`). We can optimise the "is visible" or whatever you're trying to do here at a later point.

Please dial this PR back to be just about the design.

Reply:
As discussed, there should be basic time-based debouncing instead of this kind of logic.

Reply:
I think it's fine to add that as an afterthought. It's going to be added complexity and I'd rather this PR just stays on getting the design right first.